### PR TITLE
Simplify depth cfg: drop unvalidated loss/calibration knobs, default silog_lambda=1.0

### PR DESCRIPTION
**Pull request by Bovey0809**
## Simplify depth cfg: remove unvalidated/refuted knobs, align `silog_lambda` default with released weights

Review feedback on the depth task flagged the number of depth-specific keys in `default.yaml`. This PR audits all 11 against the depth experiment records and keeps only the 5 with evidence justifying a public hyperparameter; the rest are deleted along with their code paths.

### Removed (6 keys, with the evidence)

| Key | Why removed |
|---|---|
| `silog_trim` | A/B tested at 0.1 on NYU/SUN RGB-D/KITTI (2026-07-02): harmful everywhere (NYU RMSE 0.53 → 7.06). Hypothesis refuted; shipped off by default, so a dead code path. |
| `cal_dist_pw` | 20-run calibration ablation (2026-06-25): provably inert — dist_power 0/1/2 produce identical δ1 on every dataset/model tested. |
| `dist_pw` | Never validated: the density-gated experiment arms that would test it were never run. Off by default, only active behind `dense_gate>0`. |
| `dense_gate` | Exists only to gate `dist_pw` + MiDaS-weighted coarse-grad levels; same never-run experiment. |
| `silog_l1` | No experimental record anywhere (never set nonzero in any run). Its purpose — scale anchoring under a scale-invariant loss — is served by the validated post-training calibration path. |
| `silog_grad_min_valid` | Hardcoded to 0.5 instead: it is a correctness guard (blocks noisy coarse gradients on sparse LiDAR GT), not a tuning lever; 0.5 is the only value ever validated (guard == single-scale on KITTI exactly, dense unchanged). |

### Changed

- **`silog_lambda` default 0.5 → 1.0.** The released `yolo26*-depth.pt` family was retrained and re-released at λ=1.0 after a controlled isolation run showed λ=1.0 alone accounted for a +0.025 mean-δ1 gain (and fixed the n>s size inversion). The 0.5 default was a fossil of a dead env var; with it, anyone fine-tuning the released weights silently trains with a different loss than pretraining. λ=1.0 is fully scale-invariant, which is covered by default-on `auto_calibrate` (the released checkpoints already ship with baked calibration).

### Kept (5 keys)

`silog`, `silog_grad` (loss-component gains, matching the `box`/`cls`/`dfl` convention; doubling `silog_grad` was A/B tested and regressed every benchmark, so the value is load-bearing), `silog_grad_scales` (dataset-dependent: helps dense GT, can hurt on large mixed corpora — a genuine user decision), `silog_lambda`, `auto_calibrate` (extensively validated do-no-harm policy; experiments themselves needed the off switch).

At default settings the loss is numerically unchanged except for the intended λ flip: every removed branch was inactive at its default value.

**Deleted:** cfg keys `silog_trim`, `silog_l1`, `dist_pw`, `dense_gate`, `cal_dist_pw`, `silog_grad_min_valid` (from `default.yaml` + type-check sets in `cfg/__init__.py`); the SILog trimming branch, distance-weighting branch, density-gated pyramid weighting, and scale-anchored L1 term in `v8DepthLoss`; the `dist_power` weighted-fit path threaded through `calibrate.py` (`lstsq_affine`, `select_calibration`, `select_calibration_cv`, `fit_calibration_selective`, `calibrate_checkpoint`) and its call sites in `depth/train.py`/`depth/val.py`; tests of the removed behaviors. Net −139 lines.

Old checkpoints trained with the removed keys still load: stale `train_args` entries become inert attributes and `Model._reset_ckpt_args` strips them at the facade.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🧹 Simplifies depth training and calibration by removing experimental weighting, trimming, and density-gating options while standardizing SILog settings.

### 📊 Key Changes
- 🗑️ Removed depth-loss options for:
  - Distance weighting (`dist_pw`)
  - Scale-anchored L1 loss (`silog_l1`)
  - SILog residual trimming (`silog_trim`)
  - Dense-image gating (`dense_gate`)
  - Configurable sparsity thresholds (`silog_grad_min_valid`)
- ⚙️ Updated defaults:
  - `silog_lambda` increased from `0.5` to `1.0`, matching released YOLO26 depth weights.
  - Multi-scale gradient matching remains enabled with four scales.
- 📏 Simplified SILog computation to use uniform pixel averaging and added protection against small negative values that could produce NaNs.
- 🧭 Simplified depth calibration by removing far-pixel distance weighting and related function arguments.
- ✅ Updated training, validation, plotting, and unit tests to reflect the streamlined APIs and configuration.

### 🎯 Purpose & Impact
- 🎯 Aligns training defaults with pretrained YOLO26 depth checkpoints, improving compatibility and reproducibility.
- 🚀 Reduces configuration complexity and removes experimental behaviors that could make depth training harder to tune.
- 🛡️ Improves numerical stability by preventing invalid square-root operations in the SILog loss.
- 🔄 Makes calibration behavior more predictable by using uniform least-squares fitting and retaining the existing “calibrate only if it helps” policy.
- ⚠️ Users relying on removed depth arguments will need to remove or update them in training configurations.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Bovey0809, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated review has noted your depth-configuration simplification; an Ultralytics engineer will assist with the review. Please check the following:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/utils/loss.py [hunk 1]
```diff
@@ -1248,44 +1233,10 @@ def __call__(self, preds, batch):
...
-            log_diff = log_diff[keep]
-            if torch.is_tensor(w):
-                w = w[keep]
-                w = w / w.mean()  # renormalize to mean 1 over the kept set (proper weighted mean)
-
-        def wmean(x):
-            return (w * x).mean()
-
-        silog = torch.sqrt(wmean(log_diff**2) - self.silog_lambda * wmean(log_diff) ** 2 + 1e-6)
+        silog = torch.sqrt((log_diff**2).mean() - self.silog_lambda * log_diff.mean() ** 2 + 1e-6)
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/loss.py]: 💡 **MEDIUM**: With the new default `silog_lambda=1.0`, this computes a variance as `E[x²] - E[x]²`. In float32/AMP, a near-constant log residual (for example, an initially uniform scale error) can suffer cancellation and become slightly negative by more than `1e-6`, making `sqrt` return NaN and poisoning training. Clamp the mathematically non-negative variance before the square root.

**Suggested change:**
```suggestion
        silog = torch.sqrt(((log_diff**2).mean() - self.silog_lambda * log_diff.mean() ** 2).clamp_min(0) + 1e-6)
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The configuration and calibration API removals are consistently propagated through the trainer, validator, tests, and config validation. One numerical-stability fix is needed because making fully scale-invariant SILog the default exposes cancellation in the current variance calculation.

💬 Posted 1 inline comment

File: ultralytics/utils/loss.py (hunk 1 above)
Comment (Bovey0809) [reply ultralytics/utils/loss.py]: Applied in a3fd9e613 — clamped the variance before the sqrt exactly as suggested. Verified NaN-free with a pure-scale-error residual (×2 and ×150) at λ=1.0 in fp32 and under bf16 autocast, gradients finite.

File: ultralytics/utils/loss.py [hunk 2]
```diff
@@ -1295,30 +1246,23 @@ def wmean(x):
...
             # Masked x2 average-pool: aggregate only valid pixels so invalid zeros never bleed
             # into valid neighbors (same rationale as never resizing sparse GT above).
             vp = F.avg_pool2d(valid_f, 2)
-            if not gated and vp.mean() < self.grad_min_valid:
-                break  # sparsity guard: GT too sparse for reliable coarse gradients
+            if vp.mean() < 0.5:
+                # Sparsity guard: coarse pooled cells over sparse GT (e.g. LiDAR) aggregate too
+                # few real pixels, so their gradients are noise (empirically degrades KITTI).
+                # Dense GT stays well above 0.5 valid fraction and keeps all levels.
+                break
```
[the remark this answers is not in the document]
Comment (Bovey0809) [reply ultralytics/utils/loss.py:1262]: Correct observation, but the batch-level guard is pre-existing behavior — this PR only hardcodes the previously-configurable 0.5 threshold, it doesn't change the guard's granularity. Per-image gating of coarse levels is exactly the density-gated machinery this PR deletes (`dense_gate`), which was never experimentally validated; the batch-level guard as implemented is what the released-family training runs validated (KITTI guard == single-scale exactly, dense unchanged). If the pending density-gate experiments ever show a win, per-image gating can return with evidence.

File: ultralytics/models/yolo/depth/calibrate.py [hunk 3]
```diff
@@ -212,14 +200,12 @@ def _collect_logpairs(model, dataloader, device, max_images: int):
...
-def fit_calibration_selective(
-    model, dataloader, device, max_images: int = 200, margin: float = 0.002, dist_power: float = 0.0
-):
+def fit_calibration_selective(model, dataloader, device, max_images: int = 200, margin: float = 0.002):
     """Select and apply calibration via "calibrate only if it helps" (see :func:`select_calibration_cv`).
 
     Collects per-image ``(log_pred, log_gt)`` over the loader, splits images into independent fit/score folds (no
     leakage), chooses identity / scale-only / affine by cross-validated raw-scale δ1, and writes the winner into the
-    head's ``cal_a``/``cal_b``. ``dist_power > 0`` weights the affine fit toward far pixels (see :func:`lstsq_affine`).
+    head's ``cal_a``/``cal_b``.
```
[the remark this answers is not in the document]
Comment (Bovey0809) [reply ultralytics/models/yolo/depth/calibrate.py]: Fixed in a3fd9e613 — the docstring now states affine is opt-in via `allow_affine` and not offered by `fit_calibration_selective`; also fixed the adjacent `The: func:` RST typo.

**Comment by UltralyticsAssistant** [comment]
🎉 Merged—thank you @Bovey0809, with contributions from @glenn-jocher!

> “Simplicity is the ultimate sophistication.” — Leonardo da Vinci

This PR brings that principle to YOLO26 depth training by streamlining SILog configuration, removing experimental weighting and gating options, improving numerical stability, and simplifying calibration. These changes make depth workflows more predictable, reproducible, and easier to maintain. We truly appreciate your valuable contributions!
